uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1017.php

```diff
@@ -102,4 +161,33 @@ private function up1017_convertLocationNamesToIds($locationTypeMap) {
     return $locationTypeMap;
   }
 
+  /**
+   * Looks up the location type ID for a given location type name
+   *
+   * @param $locationTypeName
+   *
+   * @return int
+   */
+  private function up1017_getLocationTypeID($locationTypeName) {
+    $locationIds = $this->up1017_getLocationTypeIds();
+
+    return (int) CRM_Utils_Array::value($locationTypeName, $locationIds);
+  }
+
+  /**
+   * Gets a list of location type names mapped to their database ID
+   *
+   * @return array
+   */
+  private function up1017_getLocationTypeIds() {
+    static $locationTypeIds = [];
```

Comment:
I think I like the use of static here. Would help avoid multiple API calls

Reply:
Yeah, ideally things like this would be a class property, but it gets more difficult with traits as property names can conflict.